### ISSUE TITLE
Vault module doesn't support file backend properly

### ISSUE THREAD

**Issue by jpotier** [NixOS/nixpkgs#72786]
**Describe the bug**
Using backend `file` in the vault module leads to a failure to build

**To Reproduce**
1. set `services.vault.storageBackend` to `"file"`
2. `nixos-rebuild build`
3. Fails with an error concerning systemd.tmpfiles.rules:
```
error: The option value `systemd.tmpfiles.rules.[definition 4-entry 1]' in `/var/src/nixpkgs/nixos/modules/services/security/vault.nix' is not of type `string'.
```

**Expected behavior**
Machine configuration builds successfully

**Additional context**
Discovered this  while trying out another backend than `inmem` for vault module.

**Metadata**
Please run `nix run nixpkgs.nix-info -c nix-info -m` and paste the result.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- services.vault.storageBackend
# a list of nixos modules affected by the problem
module:
- vault
```

**Comment by jpotier** [comment]
Now that this fix is in master, how does it reach stable?

**Comment by johanot** [comment]
@jpotier Just stumbled into this issue myself. And... For the record: The fix has reached the 19.09 channel at this time. :)

**Comment by jpotier** [comment]
> The fix has reached the 19.09 channel at this time. :)

Magic!
